Lantiq Image: allow profiles to specify the name of the related dts file
- if a variable with the syntax $(PROFILE)_DTS is defined,
  use the content as basename of the dts

diff --git a/target/linux/lantiq/image/Makefile b/target/linux/lantiq/image/Makefile
--- a/target/linux/lantiq/image/Makefile
+++ b/target/linux/lantiq/image/Makefile
@@ -37,7 +37,7 @@ endef
 
 define PatchKernelLzma
 	cp $(KDIR)/vmlinux$(2) $(KDIR)/vmlinux$(2)-$(1)
-	$(LINUX_DIR)/scripts/dtc/dtc -O dtb -o $(KDIR)/$(1).dtb ../dts/$(1).dts
+	$(LINUX_DIR)/scripts/dtc/dtc -O dtb -o $(KDIR)/$(1).dtb ../dts/$(if $(3),$(3),$(1)).dts
 	$(STAGING_DIR_HOST)/bin/patch-dtb $(KDIR)/vmlinux$(2)-$(1) $(KDIR)/$(1).dtb
 	$(call CompressLzma,$(KDIR)/vmlinux$(2)-$(1),$(KDIR)/vmlinux$(2)-$(1).lzma)
 endef
@@ -238,11 +238,11 @@ endif
 endef
 
 define Image/BuildKernel/Template
-	$(call PatchKernelLzma,$(1))
+	$(call PatchKernelLzma,$(1),,$(if $(DTS_$(1)),$(DTS_$(1))))
 	$(call MkImageLzma,$(1))
 	$(CP) $(KDIR)/uImage-$(1) $(BIN_DIR)/$(IMG_PREFIX)-$(1)-uImage
 ifneq ($(CONFIG_TARGET_ROOTFS_INITRAMFS),)
-	$(call PatchKernelLzma,$(1),-initramfs)
+	$(call PatchKernelLzma,$(1),-initramfs,$(if $(DTS_$(1)),$(DTS_$(1))))
 	$(call MkImageLzma,$(1),-initramfs)
 	$(CP) $(KDIR)/uImage-$(1)-initramfs $(BIN_DIR)/$(IMG_PREFIX)-$(1)-uImage-initramfs
 endif
